tests/testthat/testDIFFUSE.R

```diff
@@ -0,0 +1,290 @@
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG5, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG6, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG7, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG8, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG9, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+    expect_error(DIFFUSE(badAGG10, algorithm="Leis" ,
+                         param <- list(getOption("rete.beta")),
+                         silent = TRUE, noLog = FALSE))
+
+})
+
+context("Check that edge scores assigned to EGG produced in test match
+        those in manually constructed reference")
+
+testthat::test_that("EGG made correctly", {
+    EGG<-DIFFUSE(AGG, algorithm = "Leis",
+                 param <- list(getOption("rete.beta")),
+                 silent = TRUE, noLog = FALSE)
+    eggEdges <- igraph::as_data_frame(EGG, what = "edges")
+    eggRefEdges<-igraph::as_data_frame(eggRef, what = "edges")
+
+    fromLogiVect <- eggEdges$from == eggRefEdges$from
+    toLogiVect <- eggEdges$to == eggRefEdges$to
+    edgeIDLogiVect <- eggEdges$edgeID == eggRefEdges$edgeID
+    inflLogiVect <- eggEdges$Influence == eggRefEdges$Influence
+
+    expect_equal(sum(as.numeric(as.logical(fromLogiVect))), length(fromLogiVect))
+    expect_equal(sum(as.numeric(as.logical(toLogiVect))), length(toLogiVect))
+    expect_equal(sum(as.numeric(as.logical(edgeIDLogiVect))), length(edgeIDLogiVect))
+    expect_equal(sum(as.numeric(as.logical(inflLogiVect))), length(inflLogiVect))
+    # for each comparison of columns, we expect all rows to have value of true
+})
+
+#To do: have a way of checking that metadata is assigned correctly
+
+#context ("correct metadata assigned")
+#test_that("metadata correctly assigned to EGG", {
+
+#})
+
+#Below test was totally plagiarized from diffuseV3.R, but better not to
+#re-engineer the wheel
+test_that("DIFFUSE doesn't leak output if silent = TRUE", {
+    testF <- tempfile()
+    capture.output(dummy <- DIFFUSE(AGG = AGG, algorithm = "Leis",
+                   param = list(getOption("rete.beta")),
+                   silent = TRUE,
+                   noLog = FALSE),
+                   file = testF)
+    expect_equal(length(readLines(testF)), 0)
+    expect_true(file.remove(testF))
+})
+
+#To do:
+#Come up with a test to see that nothing is written to the log if noLog is TRUE
+
+#[END]
```

Comment:
How can you validate that the results are correct? I mean not in a numeric sense, but in a semantic sense. Do the numbers correctly represent heat diffusion? How can you tell?